src/wallet.ts

```diff
@@ -0,0 +1,2170 @@
+    //////////////////////////////////////////////////////////////////
+
+    // define the address button
+    let backupKeys = blessed.button({
+        parent: settingsWindow,
+        mouse: true,
+        keys: true,
+        shrink: true,
+        left: 'center',
+        top: 'center',
+        tags: true,
+        content: '{bold}click here or ctrl+c to copy private keys to the clipboard{/}',
+        style: {
+            bg: 'red',
+            fg: 'white',
+            hover: {
+                bg: 'black',
+                fg: 'red'
+            }
+        }
+    });
+
+    // on addressbutton click
+    backupKeys.on('click', async function() {
+        const [privateSpendKey, privateViewKey] = wallet.getPrimaryAddressPrivateKeys();
+        const [mnenomicSeed] = wallet.getMnemonicSeed();
+        const backupMessage = 
+            'Public address:\n' +
+            `${addressString}\n\n` +
+            'Private view key:\n' +
+            `${privateViewKey}\n\n` +
+            'Private spend key:\n' +
+            `${privateSpendKey}\n\n` +
+            'Mnemonic seed:\n' +
+            `${mnenomicSeed}`
+            
+        clipboardy.writeSync(backupMessage);
+        notifyUser(notificationText, '{bold}{red-fg}PRIVATE KEYS COPIED TO CLIPBOARD{/}', 2000);
+        settingsWindow.focus();
+    });
+
+        // on addressbutton click
+        backupKeys.on('press', async function() {
+            const [privateSpendKey, privateViewKey] = wallet.getPrimaryAddressPrivateKeys();
+            const [mnenomicSeed] = wallet.getMnemonicSeed();
+            const backupMessage = 
+                'Public address:\n' +
+                `${addressString}\n\n` +
+                'Private view key:\n' +
+                `${privateViewKey}\n\n` +
+                'Private spend key:\n' +
+                `${privateSpendKey}\n\n` +
+                'Mnemonic seed:\n' +
+                `${mnenomicSeed}`
+                
+            clipboardy.writeSync(backupMessage);
+            notifyUser(notificationText, '{bold}{red-fg}PRIVATE KEYS COPIED TO CLIPBOARD{/}', 500);
+            settingsWindow.focus();
+        });
+    
+
+
+    ///////////////////////////////////////////////////////////////////
+    //  SYNC BAR
+    //////////////////////////////////////////////////////////////////
+    let progress = blessed.progressbar({
+        parent: walletWindow,
+        style: {
+            fg: 'red',
+            bg: 'default',
+            bar: {
+                bg: 'default',
+                fg: 'red'
+            },
+            border: {
+                fg: 'default',
+                bg: 'default'
+            }
+        },
+        ch: ':',
+        width: '100%',
+        height: 1,
+        top: '90%',
+        left: '0%',
+        filled: 0
+    });
+
+    // refresh the sync information every second
+    refreshSync(syncStatus, wallet, progress);
+    setInterval(refreshSync.bind(null, syncStatus, wallet, progress), 1000);
```

Comment:
Not sure why we're using `.bind` here - that's usually reserved for binding the `this` argument to a function. I would rather see
```suggestion
    setInterval(() => refreshSync(syncStatus, wallet, progress), 1000);
```

Futhermore, we're not assigning the `setInterval` result to anything which we can then call `clearInterval` on. This will probably result in this method still being called even once the wallet is closed (an potentially another reopened), which is going to result in wasted cpu/memory and possibly cause crashes.